### PR TITLE
switch CI and publishing to Github Actions

### PR DESCRIPTION
**Pull request by razumau**
Adds workflows for testing (on Ruby 3.0 to 3.3) and publishing.
Switches to pygments 2.x (which uses Python 3).

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-require "bundler/setup"
```
Comment (grosser) [inline Rakefile:1]: should stay ?

File: Rakefile [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 require "bundler/gem_tasks"
 require "double_doc/task"
 require "rake/testtask"
-require "bump/tasks"
```
Comment (grosser) [inline Rakefile:5]: we sill need this for `rake bumpminor` etc ?

**Review by grosser — DISMISSED** [review]
document how to release a new version in the readme
something like https://github.com/zendesk/stronger_parameters?tab=readme-ov-file#releasing-a-new-version

File: Rakefile (hunk 1 above)
Comment (grosser) [reply Rakefile:1]: so it uses bundled libraries when someone runs a plain `rake foo`

File: doc/readme.md [hunk 3]
```diff
@@ -63,9 +63,13 @@ And DoubleDoc will generate this markdown document:
 @import lib/double_doc/task.rb
 
 ### Notes
- - Tested on ruby 2.0+
+ - Tested on ruby 3.0+
  - Does not work on jruby because of its dependency on redcarpet.
 
+### Release
+
+After merging your changes, create a PR with a version bump and updated changelog. After that PR gets merged, create a new tag (by running `gem_push=no rake release` or via Github releases). This will trigger the publishing workflow (which will need to be [approved in Github Actions](https://github.com/zendesk/double_doc/actions/workflows/publish.yml)).
```
Comment (grosser) [inline doc/readme.md:71]: `rake release` task is provided by `bundler/gem_tasks` which was removed ?

Comment (grosser) [inline doc/readme.md:71]: easier to follow if you break it into a list (bullet points)

File: readme.md [hunk 4]
```diff
@@ -99,9 +99,13 @@ Then run `rake double_doc`, which will generate a `readme.md` in the root of the
 If a gh-pages branch exists, run `rake doc:publish` to generate html documentation and push it to your github pages.
 
 ### Notes
- - Tested on ruby 2.0+
+ - Tested on ruby 3.0+
  - Does not work on jruby because of its dependency on redcarpet.
 
+### Release
+
+After merging your changes, create a PR with a version bump and updated changelog. After that PR gets merged, create a new tag (by running `gem_push=no rake release` or via Github releases). This will trigger the publishing workflow (which will need to be [approved in Github Actions](https://github.com/zendesk/double_doc/actions/workflows/publish.yml)).
```
Comment (grosser) [inline readme.md:107]: why do we have 2 readmes with the same content ? ... can we just put this into the main one ?

**Review by grosser — APPROVED** [review]

File: readme.md (hunk 4 above)
Comment (razumau) [reply readme.md:107]: I think the idea is that it showcases the gem: I ran `rake doc` to generate the root readme (and an html file in `./site` which is in gitignore).

**Review by grosser — APPROVED** [review]

File: readme.md (hunk 4 above)
Comment (bquorning) [reply readme.md]: The gem is literally called Double Doc, so… 😂

**Pull request merged**